spring/boot-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java

```diff
@@ -258,15 +260,12 @@ private void configureArmeriaService(ServerBuilder sb, ArmeriaSettings settings)
     }
 
     private <T> Optional<T> findBean(Class<T> clazz) {
-        final List<T> beans = findBeans(clazz);
-        switch (beans.size()) {
-            case 0:
-                return Optional.empty();
-            case 1:
-                return Optional.of(beans.get(0));
-            default:
-                throw new IllegalStateException("Too many " + clazz.getSimpleName() + " beans: " +
-                                                beans + " (expected: 1)");
+        try {
+            return Optional.of(beanFactory.getBean(clazz));
+        } catch (NoUniqueBeanDefinitionException e) {
```

Comment:
If this is finding the first one among beans, shouldn't we remove this clause and just leave catching `NoSuchBeanDefinitionException` clause?